perf/change_point_detector.go

```diff
@@ -0,0 +1,74 @@
+package perf
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/evergreen-ci/cedar"
+	"io"
+	"net/http"
+)
+
+type Detector interface {
+	DetectChanges([]float64) ([]ChangePoint, error)
+}
+
+type ChangePoint struct {
+	Index     int
+	Algorithm Algorithm
+}
+
+type Algorithm struct {
+	Name          string
+	Version       string
+	Configuration map[string]interface{}
```

Comment:
would it be possible to have Configuration use a non-map type?

Reply:
The configuration is basically dynamic (varies depending on the algorithm being used). Not sure what would be better in this case.

Reply:
shouldn't this match something that we pass into the ChangeDetector?

Reply:
Sure, but each algorithm will have its own unique configuration --> I'm not sure how best to represent that in golang on an abstract level; for a given specific algorithm we can obviously specify exactly how to configure it, but it would only be conforming to the structure/interface provided by this abstraction?

Reply:
I think this is the correct structure: 

```
type ChangePoint struct {
	Index int           `bson:"index" json:"index" yaml:"index"`
	Info  AlgorithmInfo `bson:"info" json:"info" yaml:"info"`
}

type AlgorithmInfo struct {
	Name    string            `bson:"name" json:"name" yaml:"name"`
	Version int               `bson:"version" json:"version" yaml:"version"`
	Options []AlgorithmOption `bson:"options" json:"options" yaml:"options"`
}

type AlgorithmOption struct {
	Name  string      `bson:"name" json:"name" yaml:"name"`
	Value interface{} `bson:"value" json:"value" yaml:"value"`
}
```